### PR TITLE
Convert Gradle builds to KTS

### PR DESCRIPTION
**Pull request by swankjesse**
I'm about to move the plugin into its own subproject and it's
way easier for me to work in KTS.

### CODE REVIEW

**Review by oldergod — APPROVED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  dependencies {
+    classpath("com.android.tools.build:gradle:4.2.1")
+    classpath("net.ltgt.gradle:gradle-errorprone-plugin:2.0.1")
+    classpath("com.vanniktech:gradle-maven-publish-plugin:0.14.2")
+    classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:1.5.20")
+  }
+}
+
+plugins {
+  id("com.gradle.plugin-publish") version "0.12.0" apply false
```
Comment (JakeWharton) [inline build.gradle.kts:22]:
```suggestion
```
Same as other PR, we don't use this

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+object Dependencies {
```
Comment (JakeWharton) [inline buildSrc/src/main/kotlin/Dependencies.kt:1]: Ewwwww buildSrc for dependencies is 👎

File: quickjs/android/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+dependencies {
+  api(Dependencies.androidxAnnotation)
+
+  androidTestImplementation(Dependencies.androidxTestRunner)
+  androidTestImplementation(Dependencies.truth)
+}
+
+fun quickJsVersion(): File {
```
Comment (JakeWharton) [inline quickjs/android/build.gradle.kts:72]: This needs to return a String

**Review by JakeWharton — APPROVED** [review]
Screen Shot 2021-08-18 at 11 02 59 AM

**Comment by swankjesse** [comment]
Merged manually.

File: quickjs/android/build.gradle.kts (hunk 3 above)
Comment (swankjesse) [reply quickjs/android/build.gradle.kts:72]: Oooh yuck, I replaced a file’s contents with its path. Ugh. Fixed.

File: build.gradle.kts (hunk 1 above)
Comment (swankjesse) [reply build.gradle.kts:22]: Fixed.

File: buildSrc/src/main/kotlin/Dependencies.kt (hunk 2 above)
Comment (swankjesse) [reply buildSrc/src/main/kotlin/Dependencies.kt:1]: Yeah, let’s switch to the Gradle 7 thing?

**Pull request closed without merging**